src/hyperlink/_url.py

```diff
@@ -1763,8 +1776,18 @@ class DecodedURL(object):
         lazy (bool): Set to True to avoid pre-decode all parts of the URL to
             check for validity. Defaults to False.
 
+    .. note::
+
+      The :class:`DecodedURL` initializer takes a :class:`URL` object,
+      not URL components, like :class:`URL`. To programmatically
+      construct a :class:`DecodedURL`, you can use this pattern:
+
+        >>> DecodedURL().replace(host='pypi.org', path=('projects', 'hyperlink').to_text()
+        "http://pypi.org/projects/hyperlink"
+
+    *(New in 18.0.0)*
```

Comment:
[``.. versionadded::``](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded)?

Reply:
lol I knew I was forgetting something, lemme fix that

Reply:
fixed, thanks!